### PR TITLE
Remove deprecated flags before `v18.0.0`

### LINKED ISSUE #14070: Remove deprecated flags before v18.0.0
Let's remove the following flags before v18:

- vtctld_show_topology_crud
- init_populate_metadata
- skip-preflight
- vtctld_show_topology_crud
- vtctld_addr
- durability_policy
- lock-shard-timeout
- orc_web_dir
- use_super_read_only
- disable-replication-manager
- init_populate_metadata
- queryserver-config-pool-prefill-parallelism
- queryserver-config-stream-pool-prefill-parallelism
- queryserver-config-transaction-prefill-parallelism
- queryserver-config-schema-change-signal-interval

### PR DESCRIPTION
**Pull request by frouioui**
## Description

This PR removes bunch of flags that were deprecated in v17 or before. It also removes the `status` package.

## Related Issue(s)

- Fixes #14070 

## Checklist

-   [x] "Backport to:" labels have been added if this change should be back-ported
-   [x] Tests were added or are not required
-   [x] Did the new or modified tests pass consistently locally and on the CI
-   [x] Documentation was added or is not required

### CODE REVIEW

**Review by ajm188 — APPROVED** [review]

File: changelog/18.0/18.0.0/summary.md [hunk 1]
```diff
@@ -116,6 +117,35 @@ VTGate flag:
 
 - `--schema_change_signal_user` is deprecated and will be removed in `v19.0`
 
+#### <a id="deleted-flags"/>Deleted Command Line Flags
+
+Flags in `vtcombo`:
+- `--vtctld_addr`
+
+Flags in `vtctldclient`:
+- `--skip-preflight`
```
Comment (deepthi) [inline changelog/18.0/18.0.0/summary.md:126]: This flag is specific to the ApplySchema command and should be documented that way. It's not a general vtctldclient flag.
Try `vtctldclient --help` vs `vtctldclient ApplySchema --help` to see the difference.

File: changelog/18.0/18.0.0/summary.md [hunk 2]
```diff
@@ -116,6 +117,35 @@ VTGate flag:
...
+Flags in `vtgate`:
+- `--vtctld_addr`
+
+Flags in `vttablet`:
+- `--vtctld_addr`
+- `--use_super_read_only`
+- `--disable-replication-manager`
+- `--init_populate_metadata`
+- `--queryserver-config-pool-prefill-parallelis`
+- `--queryserver-config-stream-pool-prefill-parallelis`
```
Comment (deepthi) [inline changelog/18.0/18.0.0/summary.md:137]:
```suggestion
- `--queryserver-config-stream-pool-prefill-parallelism`
```

File: changelog/18.0/18.0.0/summary.md [hunk 3]
```diff
@@ -116,6 +117,35 @@ VTGate flag:
...
+
+Flags in `vtgate`:
+- `--vtctld_addr`
+
+Flags in `vttablet`:
+- `--vtctld_addr`
+- `--use_super_read_only`
+- `--disable-replication-manager`
+- `--init_populate_metadata`
+- `--queryserver-config-pool-prefill-parallelis`
```
Comment (deepthi) [inline changelog/18.0/18.0.0/summary.md:136]:
```suggestion
- `--queryserver-config-pool-prefill-parallelism`
```

File: changelog/18.0/18.0.0/summary.md [hunk 4]
```diff
@@ -116,6 +117,35 @@ VTGate flag:
...
+- `--queryserver-config-stream-pool-prefill-parallelis`
+- `--queryserver-config-transaction-pool-prefill-parallelis`
```
Comment (deepthi) [inline changelog/18.0/18.0.0/summary.md:138]:
```suggestion
- `--queryserver-config-transaction-pool-prefill-parallelism`
```

**Review by deepthi — COMMENTED** [review]
Not sure what all the test failures are about. I can re-review once those are fixed.
For a change, all upgrade/downgrade tests are passing 🎉

File: changelog/18.0/18.0.0/summary.md (hunk 1 above)
Comment (frouioui) [reply changelog/18.0/18.0.0/summary.md:126]: Fixed via 05ac2000bd

**Review by rohit-nayak-ps — APPROVED** [review]

File: changelog/18.0/18.0.0/summary.md (hunk 1 above)
Comment (mattlord) [reply changelog/18.0/18.0.0/summary.md:128]: I'm assuming we mean skip_preflight as well? That's the one that I see mentioned in v17:
```
changelog/17.0/17.0.0/changelog.md: * ApplySchema: deprecate '--skip_preflight' flag [#10716](https://github.com/vitessio/vitess/pull/10716)
```

File: changelog/18.0/18.0.0/summary.md [hunk 5]
```diff
@@ -118,6 +119,35 @@ VTGate flag:
...
+- `--skip-preflight`
+
+Flags in `vtgate`:
+- `--vtctld_addr`
+
+Flags in `vttablet`:
+- `--vtctld_addr`
+- `--use_super_read_only`
+- `--disable-replication-manager`
+- `--init_populate_metadata`
```
Comment (mattlord) [inline changelog/18.0/18.0.0/summary.md:137]: FWIW we said that we *deleted* this flag in the v16 summary. Maybe we meant deprecated there? Or maybe that's the one that we had to temporarily resurrect... 😄

File: changelog/18.0/18.0.0/summary.md [hunk 6]
```diff
@@ -118,6 +119,35 @@ VTGate flag:
...
+- `--queryserver-config-pool-prefill-parallelism`
+- `--queryserver-config-stream-pool-prefill-parallelism`
+- `--queryserver-config-transaction-pool-prefill-parallelism`
```
Comment (mattlord) [inline changelog/18.0/18.0.0/summary.md:140]: I don't see this one ever explicitly mentioned in previous changelogs but I'm assuming it was an oversight and should be deleted because all of the related ones were deprecated.

File: go/test/endtoend/cluster/vtorc_process.go [hunk 7]
```diff
@@ -197,7 +196,7 @@ func (orc *VTOrcProcess) GetVars() map[string]any {
 
 // MakeAPICall makes an API call on the given endpoint of VTOrc
 func (orc *VTOrcProcess) MakeAPICall(endpoint string) (status int, response string, err error) {
-	url := fmt.Sprintf("http://localhost:%d/%s", orc.Port, endpoint)
+	url := fmt.Sprintf("http://127.0.0.1:%d/%s", orc.Port, endpoint)
```
Comment (mattlord) [inline go/test/endtoend/cluster/vtorc_process.go:199]: This is fine, but seems unrelated?

**Review by mattlord — APPROVED** [review]

File: go/test/endtoend/cluster/vtorc_process.go (hunk 7 above)
Comment (frouioui) [reply go/test/endtoend/cluster/vtorc_process.go:199]: unrelated indeed, let me revert

File: changelog/18.0/18.0.0/summary.md (hunk 5 above)
Comment (frouioui) [reply changelog/18.0/18.0.0/summary.md:137]: Originally it was meant to be `deprecated` in `v16.0.0`: https://github.com/vitessio/vitess/commit/4f5ab229213de475cb4d46f524f566ebf89167fa

Comment (frouioui) [reply changelog/18.0/18.0.0/summary.md:137]: and went from `deprecated` to `removed` in the release notes via: https://github.com/vitessio/vitess/commit/ebfe6fe611e241b9047df184e7060b84a2019fe2

Comment (frouioui) [reply changelog/18.0/18.0.0/summary.md:137]: but the flag has always been marked as deprecated, so all good. I can change the `v16.0.0` release notes if you think it would be useful.

File: changelog/18.0/18.0.0/summary.md (hunk 1 above)
Comment (frouioui) [reply changelog/18.0/18.0.0/summary.md:128]: I have added another section below for `skip_preflight`

File: go/vt/proto/binlogdata/binlogdata.pb.go
Comment (frouioui) [inline go/vt/proto/binlogdata/binlogdata.pb.go:1]: This file and all other protobuf files below (besides `vtctldata.pb.go` and `vtctldata_vtproto.pb.go`) have their `protoc` version change from `v3.21.3` to `v4.24.3`

Comment (frouioui) [reply go/vt/proto/binlogdata/binlogdata.pb.go:1]: i reverted this
